app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -33,36 +35,39 @@ export default ({ node }) => {
       dispatch(`subscribeToBlocks`)
     },
     async queryBlockInfo({ state, commit }, height) {
-      if (!height) {
-        commit(`notifyError`, {
-          title: `Couldn't query block`,
-          body: `No Height Provided`
+      try {
+        let blockMetaInfo = state.blockMetas[height]
+        if (blockMetaInfo) {
+          return blockMetaInfo
+        }
+        state.loading = true
+        blockMetaInfo = await new Promise((resolve, reject) => {
+          node.rpc.blockchain(
+            { minHeight: height, maxHeight: height },
+            (err, data) => {
+              if (err) {
+                reject(`Couldn't query block. ${err.message}`)
+              } else {
+                resolve(data.block_metas && data.block_metas[0])
+              }
+            }
+          )
+        })
+
+        commit(`setBlockMetas`, {
+          ...state.blockMetas,
+          [height]: blockMetaInfo
         })
-        return
-      }
-      let blockMetaInfo = state.blockMetas[height]
-      if (blockMetaInfo) {
         return blockMetaInfo
+      } catch (err) {
+        commit(`notifyError`, {
+          title: `Error fetching block information`,
+          body: err.message
+        })
+        state.loading = false
```

Comment:
i think we need to set `loading` to false outside of the catch too?

Reply:
yes!